src/openrct2/world/Footpath.cpp

```diff
@@ -1753,9 +1755,9 @@ void footpath_update_queue_chains()
             {
                 if (tileElement->GetType() != TILE_ELEMENT_TYPE_ENTRANCE)
                     continue;
-                if (tileElement->properties.entrance.type != ENTRANCE_TYPE_RIDE_ENTRANCE)
+                if (tileElement->AsEntrance()->GetEntranceType() != ENTRANCE_TYPE_RIDE_ENTRANCE)
                     continue;
-                if (tileElement->properties.entrance.ride_index != rideIndex)
+                if (tileElement->AsEntrance()->GetRideIndex() != rideIndex)
```

Comment:
I'd be nicer to call the `AsEntrance()` part only once and store its pointer in a local variable.

Reply:
Tried it here, but it got a bit fiddly. I did tackle the rest.